#! /bin/sh /usr/share/dpatch/dpatch-run

## 20_vdr-1.3.39-schedule-crash-fix.dpatch by Klaus Schmidinger <Klaus.Schmidinger@cadsoft.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fixes crash of vdr when epg contains 'NULL' in titles

@DPATCH@
diff -urNad vdr-1.3.39/epg.c /tmp/dpep.4wv6zh/vdr-1.3.39/epg.c
--- vdr-1.3.39/epg.c	2006-01-15 14:58:30.000000000 +0100
+++ /tmp/dpep.4wv6zh/vdr-1.3.39/epg.c	2006-01-20 22:23:29.323762500 +0100
@@ -421,13 +421,18 @@
   strreplace(shortText, '\x87', ' ');
   strreplace(description, '\x86', ' ');
   strreplace(description, '\x87', ' ');
-
+  
+  if (!title) {
+     // we don't want any "(null)" titles
+     title = strcpyrealloc(title, tr("No title"));
+     EpgBugFixStat(12, ChannelID());
+  }
+  
   if (Setup.EPGBugfixLevel == 0)
      return;
 
   // Some TV stations apparently have their own idea about how to fill in the
   // EPG data. Let's fix their bugs as good as we can:
-  if (title) {
 
      // Some channels put the ShortText in quotes and use either the ShortText
      // or the Description field, depending on how long the string is:
@@ -609,12 +614,6 @@
             }
         }
      }
-  else {
-     // we don't want any "(null)" titles
-     title = strcpyrealloc(title, tr("No title"));
-     EpgBugFixStat(12, ChannelID());
-     }
-}
 
 // --- cSchedule -------------------------------------------------------------
 
